CMakeLists.txt

```diff
@@ -24,6 +24,18 @@ else()
     message(FATAL_ERROR "Unrecognized platform: ${CMAKE_SYSTEM_NAME}")
 endif()
 
+if(${CMAKE_CXX_COMPILER} MATCHES "x86/cl.exe$")
+    set(ARCH "x86")
```

Comment:
is this change specific to new dependencies? Can we get a comment here calling out what its for?

Reply:
Looks like this was moved from the napi project? Do we need this logic in more than one place? Also, I'm not sure this is generic enough to be at the top.

Reply:
I moved it up so that I could have the ARCH variable to download node for llhttp in Dependencies/CMakeLists.txt
I see now though that this logic definitely doesn't belong at the top, but now I'm trying to figure out how to get the ARCH without just copying the logic...

Reply:
I moved the logic into a function in `Dependencies/CMakeExtensions/CMakeLists.txt`. It's slightly nicer, but doesn't feel perfect. Any suggestions on how to improve this would be appreciated.